cou/utils/juju_utils.py

```diff
@@ -383,7 +416,19 @@ async def get_applications(self) -> dict[str, Application]:
                 series=status.series,
                 subordinate_to=status.subordinate_to,
                 units={
-                    name: Unit(name, machines[unit.machine], unit.workload_version)
+                    name: Unit(
+                        name,
+                        machines[unit.machine],
+                        unit.workload_version,
+                        [
+                            SubordinateUnit(
+                                subordinate_name,
+                                unit_charm_mapping[subordinate_name],
```

Comment:
isn't possible to just use `model.applications[subordinate_name].charm` ?

Than it's not necessary to use `_get_applications` and this dict that you created

Reply:
No, the `subordinate_name` is the unit name, like `app/0`.
Also the data is already been fetch to local. Compare to handling the unit-name to application-name string transformer, there is no different.

Reply:
Resolved because it's done.